app/src/main/java/keystoretest/com/keystoretest/SecretKeyWrapper.java

```diff
@@ -63,11 +63,11 @@ public static boolean isAliasInKeystore(String alias) throws GeneralSecurityExce
         return keyStore.containsAlias(alias);
     }
 
-    public static void deleteAlias(String alias) throws GeneralSecurityException {
+    public static boolean deleteAlias(String alias) throws GeneralSecurityException {
         final KeyStore keyStore = getKeystore();
         boolean aliasExists = keyStore.containsAlias(alias);
         keyStore.deleteEntry(alias);
-        boolean aliasExitsAfterDelete = keyStore.containsAlias(alias);
+        return keyStore.containsAlias(alias);
```

Comment:
This should probably throw an exception rather than return the boolean at this point

Reply:
Not sure what you mean by this, but would the following be better?

boolean aliasExitsAfterDelete = keyStore.containsAlias(alias);
return aliasExitsAfterDelete;

Reply:
No, more like 

```
if(keyStore.containsAlias(alias))
    throw new IllegalStateException("Failed to delete key");
```

and probably drop the return bool completely